### DISCUSSION TITLE
What is trino's memory management mechanism？

### DISCUSSION THREAD

**Discussion by jiangjiguang** [trinodb/trino#17437] [Q&A]
I am very confused on trino's memory management mechanism, 
If I have given 16G memory by  -Xmx16G in jvm.config, trino how to use theses memory.
Do trino keep theses memory as static ?  it means put the 16G memory in a pool,  and the program has to get memory from the pool ?

**Comment by hashhar** [comment]
The `-Xmx16G` specifies max heap size for the JVM. JVM allocates all of it in one go from the OS since requesting memory multiple times causes slowness and can also lead to failure if you've overcommitted.

That entire 16GB is then available for Trino to use. In addition to that some libraries may use off-heap (native) memory as well.

I'd suggest you to read about JVM memory model and management since this is not something specific to Trino itself.

**Reply by jiangjiguang** [reply]
@hashhar thanks you reply.
I am familiar with  JVM memory model.
My point is trino how to use the heap memory.  Does trino put the memory to a pool ?  the program has to get memory from the pool ?
